readthedocs/rtd_tests/tests/test_config_integration.py

```diff
@@ -75,38 +75,45 @@ def setUp(self):
 
     @mock.patch('readthedocs.doc_builder.config.load_config')
     def test_python_supported_versions_default_image_1_0(self, load_config):
+        from doc_builder.constants import DOCKER_IMAGE_SETTINGS
+
         load_config.side_effect = create_load()
         self.project.container_image = 'readthedocs/build:1.0'
         self.project.enable_epub_build = True
         self.project.enable_pdf_build = True
         self.project.save()
         config = load_yaml_config(self.version)
-        self.assertEqual(load_config.call_count, 1)
-        load_config.assert_has_calls([
-            mock.call(
+
+        expected_env_config = {
+            'allow_v2': mock.ANY,
+            'build': {'image': 'readthedocs/build:1.0'},
+            'output_base': '',
+            'name': mock.ANY,
+            'defaults': {
+                'install_project': self.project.install_project,
+                'formats': [
+                    'htmlzip',
+                    'epub',
+                    'pdf'
+                ],
+                'use_system_packages': self.project.use_system_packages,
+                'requirements_file': self.project.requirements_file,
+                'python_version': 2,
+                'sphinx_configuration': mock.ANY,
+                'build_image': 'readthedocs/build:1.0',
+                'doctype': self.project.documentation_type,
+            },
+        }
+
+        img_settings = DOCKER_IMAGE_SETTINGS.get(self.project.container_image, None)
+        if img_settings:
+            expected_env_config.update(img_settings)
+            expected_env_config['DOCKER_IMAGE_SETTINGS'] = img_settings
```

Comment:
I don't follow why we are updating the `expected_env_config` twice. Why is that?

Reply:
the value of `img_settings` is: `{'python': {'supported_versions': [2, 2.7, 3, 3.4]}}`.
and the actual call has both of the values.
Actual Call:
```python3
{
    ...
    ...
    'python': {'supported_versions': [2, 2.7, 3, 3.4]},
    ...
    ...
    'DOCKER_IMAGE_SETTINGS': {'python': {'supported_versions': [2, 2.7, 3, 3.4]}},
    ...
    ...
}

Reply:
Looks like we need to refactor the code to only have/use `DOCKER_IMAGES_SETTINGS`

Reply:
@stsewd 

https://github.com/rtfd/readthedocs.org/blob/a071a81a06edb013c4a8eb91b97563f5cf7e52d6/readthedocs/doc_builder/config.py#L60-L62

I just removed the `line number 61` and all the tests pass in local. I am not very familiar with this part of code. So can you please tell me if that is all required? or it will break some things in production?

Reply:
Looks like this was added in https://github.com/rtfd/readthedocs.org/pull/3339. I'm investigating more, but it looks like it was for doing what we are doing now.

Reply:
I just opened https://github.com/rtfd/readthedocs.org/pull/5116 to have this more clear, after that PR is merged, it should be easier to see what parts of the code remove/replace.

Reply:
Thank you @stsewd 
I will update this PR after merging of #5116